packages/core/src/types.ts

```diff
@@ -1043,6 +1043,14 @@ export type MachineConfig<
     ? { context?: InitialContext<LowInfer<TContext>, TInput> }
     : { context: InitialContext<LowInfer<TContext>, TInput> });
 
+export type MachineStates<Literal = string> = {
+  states?: Record<string, MachineStates<Literal>>;
+  on?: unknown;
+  invoke?: SingleOrArray<{
+    src: Literal | NonReducibleUnknown;
+  }>;
```

Comment:
this fixes the contextual type for `invoke.src` so it can be retained as a string literal